backend/api/views.py

```diff
@@ -16,6 +17,45 @@ class RoadSegmentViewSet(viewsets.ModelViewSet):
     permission_classes = (permissions.IsAuthenticatedOrReadOnly, IsAdminOrReadOnly,)
 
 
+class ProductionDataViewSet(viewsets.ModelViewSet):
+    """
+    This viewset supports `create` and `list` actions.
+    """
+    queryset = ProductionData.objects.all()
+    serializer_class = ProductionDataSerializer
+    # Only registered users can use this view
+    permission_classes = (permissions.IsAuthenticated, IsStaffOrCreateOnly,)
+
+    def create(self, request, *args, **kwargs):
+        """
+        Add support for creating when the input data is a list
+        """
+        many = False
+
+        # Check if the incoming data is a list
+        # If it is a list set the many flag to True
+        if isinstance(request.data, list):
+            many = True
+            # Remove below if no limit on input data is needed
+            if len(request.data) > 100000:  # Change this number to the desired input size limit
```

Comment:
This is sort of a magic number. We can save the number as a constant value somewhere in our settings, maybe a new file called constants.py so it's separated out from the rest. Then, you should be able to import it to this file and implement it.